sig-architecture/production-readiness.md

```diff
@@ -15,63 +15,121 @@ aspects of the process.
-    to limitations for users)?
-  - Will enabling / using the feature result in increasing size or count
-    of the existing API objects?
-  - Will enabling / using the feature result in increasing time taken
-    by any operations covered by [existing SLIs/SLOs][] (e.g. by adding
-    additional work, introducing new steps in between, etc.)?
-    Please describe the details if so.
-  - Will enabling / using the feature result in non-negligible increase
-    of resource usage (CPU, RAM, disk IO, ...) in any components?
-    Things to keep in mind include: additional in-memory state, additional
-    non-trivial computations, excessive access to disks (including increased
-    log volume), significant amount of data sent and/or received over
-    network, etc. Think through this in both small and large cases, again
-    with respect to the [supported limits][].
-* Rollout, Upgrade, and Rollback Planning
-* Dependencies
-  - Does this feature depend on any specific services running in the cluster
-    (e.g., a metrics service)?
-  - How does this feature respond to complete failures of the services on
-    which it depends?
-  - How does this feature respond to degraded performance or high error rates
-    from services on which it depends?
-* Monitoring requirements
-  - How can an operator determine if the feature is in use by workloads?
-  - How can an operator determine if the feature is functioning properly?
-  - What are the service level indicators an operator can use to determine the
-    health of the service?
-  - What are reasonable service level objectives for the feature?
-* Troubleshooting
-  - What are the known failure modes?
-  - How can those be detected via metrics or logs?
-  - What are the mitigations for each of those failure modes?
-  - What are the most useful log messages and what logging levels do they require?
-  - What steps should be taken if SLOs are not being met to determine the
-    problem?
+#### Feature enablement and rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate
+	  - Feature gate name:
+    - Components depending on the feature gate:
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**  
+  Describe the consequences on existing workloads (e.g. if this is runtime
+  feature, can it break the existing applications?).
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+* **Are there any tests for feature enablement/ disablement?**  
+  TODO(wojtek-t): Do we even have any such tests for existing features currently?
```

Comment:
We should have conversion tests for versions of resources, at a minimum.

Reply:
Added clarification